packages/modular-scripts/package.json

```diff
@@ -90,6 +92,8 @@
     "postcss-preset-env": "6.7.0",
     "postcss-safe-parser": "5.0.2",
     "prompts": "2.4.2",
+    "react": "17.0.2",
```

Comment:
Where do these `react` dependencies come from? We definitely don't want to be shipping a version of `react` / `react-dom` with `modular-scripts`!

Reply:
On second thoughts - can we make these `peerDependencies` for `"^16 || ^17"`.

Reply:
It's used throughout the apps under `modular-scripts/types/` - specifying `react` in `dependencies` wouldn't mean it's shipped with `modular`, just that if you `yarn add modular-scripts` without also adding `react` it would be there too... agreed that `peerDependencies` makes more sense though!

Reply:
> On second thoughts - can we make these `peerDependencies` for `"^16 || ^17"` - that's a more accurate representation of them.

@LukeSheard that's tough - peer dependencies mean "something that you need to install in the main package using this plugin / library", which makes sense in a way (if you're using modular, you should definitely install React!) but doesn't in another (this is not a plugin or library that you use at runtime, and anyway React is used only in tests, so it should be a dev dependency).

To be super pedantic, I would put react in dev deps and relax the [rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md#options) to not raise an error in certain directories, like `test`, but if you think it's too cumbersome I'm ok with `peerDependencies`.

Reply:
My preference would be to have `react` in `devDependencies` (ie: if you're building/testing `modular` itself, you will get that version of `react` installed), and as a guide to consumers you could list `react` with the broad version in `peerDependencies`, but only if `modular` would always be used in `react` based monorepos.

Due to the way `react` is imported, no errors are thrown by having it specified in `devDependencies`.

Reply:
> Due to the way `react` is imported, no errors are thrown by having it specified in `devDependencies`

Why? I'm not incredibly familiar with the rule, but I understood that with `"devDependencies": false` it will error if you import `react` (and we do in `modular-scripts`).

Reply:
Turns out I didn't see the overrides, which do exactly what I meant two comments ago :) - thanks @steveukx - @LukeSheard if ok for you, I'll close this thread.